lib/library.rb

```diff
@@ -0,0 +1,68 @@
+require 'date'
+require 'yaml'
+
+class Library
+  attr_accessor :items
+
+  def initialize
+    @items = YAML.load_file('./lib/data.yml') || []
+  end
+
+  def check_out(item, user)
+  case
+    when book_shelf_okay?(user) then perform_check_out(item, user)
+    else
+        { message: 'You have books that are overdue!' }
+    end
+  end
+
+  def check_in(item, user)
+    set_item_status(item, available: true, return_date: nil)
+    update_yaml_file
+    user.book_shelf.delete check_out_message(item)
+  end
+
+  def find(args={})
+    @items.detect{|obj| obj[:item][args.keys.first] == (args[args.keys.first]).to_s }
+  end
+
+  private
+
+  def set_today
+    Date.today.strftime('%F')
+  end
+
+  def set_return_date
+    Date.today.next_month.strftime('%F')
+  end
+
+  def set_item_status(item, args={})
+    @items.detect do |i|
+      if i[:item][:title] == item[:item][:title]
+        i[:available] = args[:available]
+        i[:return_date] = args[:return_date]
+      end
+    end
+  end
+
+  def perform_check_out(item, user)
+    set_item_status(item, available: false, return_date: set_return_date)
+    update_yaml_file
+    user.book_shelf.push check_out_message(item)
+  end
+
+  def check_out_message(item)
+    { title: item[:item][:title],
+      author: item[:item][:author],
+      checked_out_date: set_today,
+      return_date: set_return_date }
+  end
+
+  def book_shelf_okay?(user)
+    user.book_shelf.select{|a| a[:return_date] < Date.today.strftime('%F')}.empty?
```

Comment:
Metrics/LineLength: Line is too long. [82/80] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)